be/src/storage/lake/lake_primary_index.cpp

```diff
@@ -47,6 +48,38 @@ Status LakePrimaryIndex::_do_lake_load(Tablet* tablet, const TabletMetadata& met
     auto pkey_schema = ChunkHelper::convert_schema(*tablet_schema, pk_columns);
     _set_schema(pkey_schema);
 
+    // load persistent index if enable persistent index meta
+    size_t fix_size = PrimaryKeyEncoder::get_encoded_fixed_size(pkey_schema);
+
+    if (tablet->get_enable_persistent_index(base_version) && (fix_size <= 128)) {
+        DCHECK(_persistent_index == nullptr);
+
+        // Even if `enable_persistent_index` is enabled,
+        // it may not take effect because `storage_root_path` is not set
+        if (StorageEngine::instance()->is_lake_persistent_index_dir_inited()) {
+            auto persistent_index_type = tablet->get_persistent_index_type(base_version);
+            if (persistent_index_type.ok()) {
+                switch (persistent_index_type.value()) {
+                case PersistentIndexTypePB::LOCAL: {
+                    std::string path = strings::Substitute(
+                            "$0/$1/",
+                            StorageEngine::instance()->get_persistent_index_store()->get_persistent_index_path(),
+                            tablet->id());
+                    StorageEngine::instance()->get_persistent_index_store()->create_dir_if_path_not_exists(path);
+                    _persistent_index = std::make_unique<LakePersistentIndex>(path);
+                    return ((LakePersistentIndex*)_persistent_index.get())
+                            ->load_from_lake_tablet(tablet, metadata, base_version, builder);
```

Comment:
yeap

Reply:
I don't think it is good strategy to use full memory index here because if memory isn't enough, BE will oom. How about FE decide if we can create persistent index only if local disk is enable in BE?